### PR TITLE
chore: replace deprecated LegacyVersion with own one

### PR DESCRIPTION
**Pull request by gruebel**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

- the deprecated LegacyVersion was finally removed, therefore I copied the original implementation and created a drop in replacement
- after merging PR #4033 I will update the dependencies

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Eliran-Turgeman — APPROVED** [review]
☃️

File: checkov/common/packaging/version.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    @property
+    def is_postrelease(self) -> bool:
+        return False
+
+    @property
+    def is_devrelease(self) -> bool:
+        return False
+
+
+_legacy_version_component_re = re.compile(r"(\d+ | [a-z]+ | \.| -)", re.VERBOSE)
```
Comment (matansha) [inline checkov/common/packaging/version.py:89]: are the spaces part of the regular expression? 
also, i suggest to add flag for case insensitive

File: checkov/common/packaging/version.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    @property
+    def public(self) -> str:
+        return self._version
+
+    @property
+    def base_version(self) -> str:
+        return self._version
+
+    @property
```
Comment (matansha) [inline checkov/common/packaging/version.py:52]: Why do you need all this meaningless properties?

Comment (gruebel) [reply checkov/common/packaging/version.py:52]: so it is equivalent with the `Version` class inside the `packaging`  library

File: checkov/common/packaging/version.py (hunk 1 above)
Comment (gruebel) [reply checkov/common/packaging/version.py:89]: the spaces don't seem to do anything for this use case, just kept them as they were. but will add ignorecase to it

Comment (matansha) [reply checkov/common/packaging/version.py]: Are you sure? the space are meaningful in regex. did you check this? if yes, i will approve it

Comment (gruebel) [reply checkov/common/packaging/version.py]: yeap, I tested it. It didn't change the output with or without the spaces.

Comment (gruebel) [reply checkov/common/packaging/version.py]: in general you are right that spaces are meaningful, but not for what it is used here

**Review by matansha — APPROVED** [review]

**Pull request merged**